lib/tailwind/src/components/BaseFormRoot.tsx

```diff
@@ -0,0 +1,67 @@
+import { resolveSchema } from "@nanhtu/utils";
+import z from "zod";
+import { FormRootProps } from "./templates";
+import { FormState, useFormContext, useRenderTemplate } from "..";
+import React from "react";
+
+export const BaseFormRoot: React.FC<FormRootProps> = ({
+  onSubmit,
+  onError,
+  liveValidate,
+}) => {
+  // const readonly = useFormContext((state: FormState) => state.readonly);
+  const schema = useFormContext((state: FormState) => state.schema);
+  const formData = useFormContext((state: FormState) => state.formData);
+  const setErrors = useFormContext((state: FormState) => state.setErrors);
+  const RenderTemplate = useRenderTemplate();
+
+  const resolvedSchema = resolveSchema(schema);
```

Comment:
use `useMemo` hook for this to prevent recalculate schema

Reply:
updated